### PR TITLE
GEODE-6481: add tests to verify create jdbc mapping should get correc…

### PR DESCRIPTION
**Pull request by gesterzhou**
…t field type

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java [hunk 1]
```diff
@@ -2397,7 +2397,8 @@ private void processInitParams(String value) {
             addExcludePattern(split[0], excludePattern);
           }
         }
-        classPatterns.add(Pattern.compile(split[0]));
+        String classPattern = split[0].replaceAll("\\$", "\\.");
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java]: This change does not look safe to me. Out public contract with the ReflectionBasedAutoSerializer is that users can give it regular expressions that will be used to match class names. In a java regular expression "\$" would mean match a literal "$" character. You are changing it to say match a literal ".". I think your fix to support inner classes is at too low a level. It should have instead been in the code that creates a pattern in the precondition function.
I could be wrong about this. Lets talk about it.

File: geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java [hunk 2]
```diff
@@ -834,6 +889,34 @@ public void createMappingsWithExistingPdxName() {
...
+  @Test
+  public void createMappingsDoesNotRequirePdxSerializable() {
+    String region1Name = "region1";
+    setupReplicate(region1Name);
+
+    CommandStringBuilder csb = new CommandStringBuilder(CREATE_MAPPING);
+    csb.addOption(REGION_NAME, region1Name);
+    csb.addOption(DATA_SOURCE_NAME, "connection");
+    csb.addOption(TABLE_NAME, "employeeNumeric");
+    csb.addOption(PDX_NAME, EmployeeNumeric.class.getName());
```
Comment (dschneider-pivotal) [inline geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java]: try changing 'EmployeeNumeric.class.getName()' to: "org.apache.geode.connectors.jdbc.internal.cli.CreateMappingCommandDUnitTest.EmployeeNumeric"
Does that cause it to work without your change of \\$ to \\.?

Comment (gesterzhou) [reply geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java]: This does not work, but this.reflectionBasedAutoSerializerFactory.create("\\Q" + className + "\\E"); will work.

File: geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java [hunk 3]
```diff
@@ -775,6 +782,54 @@ public void fromData(PdxReader reader) {
     }
   }
 
+  public static class EmployeeNumeric implements PdxSerializerObject {
```
Comment (jchen21) [inline geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java]: Why do we need to implement `PdxSerializerObject`?

**Pull request merged**